File: day1/Python/zeroend.py
class ZeroEnd:
    def __init__(self, n, a):
        self.__n = n
        self.__a = a
    def iszero(self):
        return 0 in self.__a
    def zeroend(self):
        count = 0
        for i in range(0, self.__n):
            if self.__a[i] != 0:
                self.__a[count] = self.__a[i]
                count += 1
        while count < self.__n:
            self.__a[count] = 0
            count += 1
        print("Array after pushing zeros to the end:")
        for i in range(0, self.__n):
            print(self.__a[i], end=" ")
def main():
    n = int(input("Enter the number elements:"))
    li = []
    for i in range(0, n):
        li.append(int(input("Enter element:")))
    obj = ZeroEnd(n, li)
    if obj.iszero():
        obj.zeroend()
    else:
        print("Array does not contain zero")
if __name__ == '__main__':
    main()

File: day1/java/zeroend.java

import java.util.*;

class Zeroarray {

    private final int n;
    private final int[] arr;

    Zeroarray(int n, int[] arr) {
        this.n = n;
        this.arr = arr;
    }

    boolean iszero() {
        for (int i = 0; i < n; i++) {
            if (arr[i] == 0) {
                return true;
            }
        }
        return false;
    }

    void pushend() {
        int j = 0;
        for (int i = 0; i < n; i++) {
            if (arr[i] != 0) {
                arr[j++] = arr[i];
            }
        }
        while (j < n) {
            arr[j++] = 0;
        }
    }

    void display() {
        System.out.println("Array after pushing zeros to the end: " + Arrays.toString(arr));
    }
}

class Zeroend {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.print("Enter the number of elements:");
        int n = scan.nextInt();
        int arr[] = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = scan.nextInt();
        }
        Zeroarray obj = new Zeroarray(n, arr);
        if (obj.iszero()) {
            obj.pushend();
            obj.display();
        } else {
            System.out.println("The given array doesn't contains any zero");
        }
    }
}
